scipy/optimize/minpack.h

```diff
@@ -73,18 +72,22 @@ the result tuple when the full_output argument is non-zero.
   /* Set up callback functions */ \
   if (!PyCallable_Check(fun) || (Dfun != Py_None && !PyCallable_Check(Dfun))) \
     PYERR(errobj,"The function and its Jacobian must be callable functions."); \
-  multipack_python_function = fun; \
-  multipack_extra_arguments = arg; \
-  multipack_python_jacobian = Dfun; \
-  multipack_jac_transpose = !(col_deriv);}
+  if (init_jac_callback(&callback, &jac_callback_info, fun, Dfun, arg, col_deriv) != 0) \
+    PYERR(errobj,"Could not init callback");\
+  callback_inited = 1; \
+  if (setjmp(callback.error_buf) != 0) \
+    PYERR(errobj,"Could not setjmp callback error_buf");\
+}
 
-#define RESTORE_JAC_FUNC() multipack_python_function = store_multipack_globals[0]; \
-  multipack_extra_arguments = store_multipack_globals[1]; \
-  multipack_python_jacobian = store_multipack_globals[2]; \
-  multipack_jac_transpose = store_multipack_globals3;
+#define RESTORE_JAC_FUNC() { \
```

Comment:
The usual idiom for C block macros is `do { ... } while(0)`

Ditto below

Reply:
I've just converted the 4 macros in minpack.h that I've actually touched to this style.